src/System.IO.Compression.Brotli/System/IO/Compression/Brotli.cs

```diff
@@ -64,20 +120,61 @@ internal static TransformationStatus Compress(ReadOnlySpan<byte> source, Span<by
                     bufOut = new IntPtr(outBytes);
                     nuint written = (nuint)destination.Length;
                     nuint consumed = (nuint)source.Length;
-                    if (!BrotliNative.BrotliEncoderCompress(quality, windowSize, encMode, consumed, bufIn, ref written, bufOut))
+                    if (!BrotliNative.BrotliEncoderCompressStream(state.BrotliNativeState, operation, ref consumed, ref bufIn, ref written, ref bufOut, out nuint totalOut))
                     {
                         return TransformationStatus.InvalidData;
-                    };
+                    }
                     bytesConsumed = (int)consumed;
                     bytesWritten = (int)written;
-                    return TransformationStatus.Done;
                 }
+                bytesWritten = destination.Length - bytesWritten;
```

Comment:
because bytesWritten after calling a native func is: "how many bytes have already free", and we convert it to how many bytes have already written